[quant][graphmode][fx][fix] Speedup all_node_args_have_no_tensors

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55352 [quant][graphmode][fx][fix] Speedup all_node_args_have_no_tensors**

Summary:
Adding a cache for this function to speedup the computation

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27584279](https://our.internmc.facebook.com/intern/diff/D27584279)

jerryzh168 committed:
[quant][graphmode][fx][fix] Speedup all_node_args_have_no_tensors

Summary:
Adding a cache for this function to speedup the computation

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx][fix] Speedup all_node_args_have_no_tensors"


Summary:
Adding a cache for this function to speedup the computation

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on torch/quantization/fx/utils.py:
```diff
@@ -346,31 +346,34 @@ def create_qparam_nodes(quantizer: QuantizerCls, node_name: str, scale: Any, zer
 
 
-def all_node_args_have_no_tensors(node: Node, modules: Dict[str, torch.nn.Module]) -> bool:
+def all_node_args_have_no_tensors(node: Node, modules: Dict[str, torch.nn.Module], cache: Optional[Dict[Node, bool]] = None) -> bool:
```
optional: we could also do something like

```
def all_node_args_have_no_tensors(node, modules, cache):
  if node in cache:
    return cache[node]
  result = _all_node_args_have_no_tensors_impl(node, modules)
  cache[node] = result
  return result
```

either is good

jerryzh168 replied:
thanks, this looks cleaner and less error prone

jerryzh168 replied:
actually, I think we still need to modify the original function to include cache..

vkuzo reviewed (approved):
thanks for speeding this up!  Just curious, what was the magnitude of the speedup?

jerryzh168 committed:
Update on "[quant][graphmode][fx][fix] Speedup all_node_args_have_no_tensors"


Summary:
Adding a cache for this function to speedup the computation

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27584279](https://our.internmc.facebook.com/intern/diff/D27584279)

[ghstack-poisoned]

jerryzh168 commented:
landed in another PR
